fendermint/vm/interpreter/src/fvm/gas/mod.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::fvm::state::FvmExecState;
+use fvm_ipld_blockstore::Blockstore;
+
+pub mod eip1559;
+
+pub type Gas = u64;
+
+/// The gas market for fendermint. This should be backed by an fvm actor.
+pub trait GasMarket {
+    /// The gas market state
+    type State;
```

Comment:
So the more I think about it, I think this trait should be state-type agnostic. The GasMarket should just have its own standard data model with its own struct, and the GasMarket implementation should adapt the concrete gas policy's details to that standard data model / interface.

Reply:
Is it possible for the trait implementor to keep a reference to `FvmExecState` instead of having to pass it in? I think you can have a `new(&FvmExecState)` trait method, with the lifetime of the GasMarket conditioned to that of the exec state?

Reply:
Or maybe it's not possible without an Rc, not sure.

Reply:
I have removed the `State` in `GasMarket`, it should be state type agnostic, technically no need to know about state and unnecessary.

I tried holding the `&mut FvmExecState` does not seem doable, because `FvmExecState<DB>` is destructed after each `deliver_tx`, then the `GasMarket` needs to be persisted every txn, which seems unnecessary. 

Technically there is no need to be aware of the `ChainState` for gas, but I also think this is part of a `ExecutionContext` or something like that to help determine the gas, implementation could just ignore.

Reply:
> I tried holding the `&mut FvmExecState` does not seem doable, because `FvmExecState<DB>` is destructed after each `deliver_tx`, then the `GasMarket` needs to be persisted every txn, which seems unnecessary.

Are you sure? `FvmExecState` seems to be maintained throughout `deliver_tx`, unless there's some nuance I'm unaware of? Did you try to make `gas_market` a member of `FvmExecState` instead of `FvmMessageInterpreter`?